src/backend/distributed/utils/statistics_collection.c

```diff
@@ -65,6 +65,17 @@ CollectBasicUsageStatistics(void)
 	memset(&unameData, 0, sizeof(unameData));
 
 	StartTransactionCommand();
+
+	/*
+	 * If there is a version mismatch between loaded version and available
+	 * version, metadata functions will fail. We return early to avoid crashing.
+	 * This can happen when updating the Citus extension.
+	 */
+	if (!CheckCitusVersion(LOG_SERVER_ONLY))
```

Comment:
Using `LOG_SERVER_ONLY` to not pollute test outputs.

Reply:
can we move this check before ```StartTransactionCommand()``` and not call ```CommitTransactionCommand()```  ?

Reply:
We cannot access catalog tables outside transactions. `CheckCitusVersion()` accesses `pg_extension`.

Reply:
ok

Reply:
:-P I was going to have the same question.